hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/WriteStatus.java

```diff
@@ -76,10 +78,15 @@ public class WriteStatus implements Serializable {
   private final boolean trackSuccessRecords;
   private final transient Random random;
 
-  public WriteStatus(Boolean trackSuccessRecords, Double failureFraction) {
+  public WriteStatus(Boolean trackSuccessRecords, Double failureFraction, Boolean isMetadataTable) {
```

Comment:
can we use primitives here to avoid developers from passing in null for `Boolean` and getting a runtime NPE?

Reply:
@lokeshj1703 : can you follow up on this. I remember hitting some issue w/ reflection or something.

Reply:
<img width="890" alt="image" src="https://github.com/user-attachments/assets/9989d509-dd5e-4eb0-9129-8e92e1847360" />
 
So, the object values are already interpreted as "Boolean" class. and the constructor args is derived as "Boolean" 

<img width="1461" alt="image" src="https://github.com/user-attachments/assets/62497045-e646-4f98-a907-573a0443a0a1" />

not sure if it can make fixes just for this. Only option I can think of is to change any "Boolean" to primitive boolean while going via reflection utils.loadClass().

which I feel we can take it up separately. 
@the-other-tim-brown : what do you think.